synapse/cortex.py

```diff
@@ -875,6 +883,33 @@ class Cortex(s_cell.Cell):
         stormvars = await self.hive.open(('cortex', 'storm', 'vars'))
         self.stormvars = await stormvars.dict()
 
+    async def _initCoreAxon(self):
+
+        self.axon = None
+        self.axready = asyncio.Event()
+
+        turl = self.conf.get('axon')
+        if turl is None:
+            path = os.path.join(self.dirn, 'axon')
+            self.axon = await s_axon.Axon.anit(path)
+            self.onfini(self.axon.fini)
+            self.axready.set()
+            return
+
+        async def teleloop():
+            self.axready.clear()
+            while not self.isfini:
+                try:
+                    self.axon = await s_telepath.openurl(turl)
+                    self.axon.onfini(teleloop)
+                    self.axready.set()
+                    return
+                except Exception as e:
```

Comment:
can't catch Exception on its own here - have to separate out asyncio.cancelled error for safe teardown.